src/result/mod.rs

```diff
@@ -26,81 +21,51 @@ mod status;
 pub use self::status::Status;
 
 /// Return type of most UEFI functions. Both success and error payloads are optional.
-pub type Result<Output = (), ErrData = ()> =
-    core::result::Result<Completion<Output>, Error<ErrData>>;
+pub type Result<Output = (), ErrData = ()> = core::result::Result<Output, Error<ErrData>>;
 
-/// Extension trait for Result which helps dealing with UEFI's warnings
+/// Extension trait which provides some convenience methods for [`Result`].
 pub trait ResultExt<Output, ErrData: Debug> {
     /// Extract the UEFI status from this result
     fn status(&self) -> Status;
 
-    /// Ignore warnings, keeping a trace of them in the logs
-    fn log_warning(self) -> core::result::Result<Output, Error<ErrData>>;
-
-    /// Expect success without warnings, panic otherwise
-    fn unwrap_success(self) -> Output;
-
-    /// Expect success without warnings, panic with provided message otherwise
-    fn expect_success(self, msg: &str) -> Output;
-
-    /// Expect error, panic with provided message otherwise, discarding output
-    fn expect_error(self, msg: &str) -> Error<ErrData>;
-
-    /// Transform the inner output, if any
-    fn map_inner<Mapped>(self, f: impl FnOnce(Output) -> Mapped) -> Result<Mapped, ErrData>;
-
     /// Transform the ErrData value to ()
     fn discard_errdata(self) -> Result<Output>;
 
-    /// Treat warnings as errors
-    fn warning_as_error(self) -> core::result::Result<Output, Error<ErrData>>
+    /// Calls `op` if the result contains a warning, otherwise returns
+    /// the result unchanged.
```

Comment:
It seems like this method should:
  - Say what the default behavior is (i.e. that warnings are treated as errors)
  - Mention that users shouldn't _normally_ call this function (i.e. it's reserved for special cases)
  - Mention the similarities to [`Result::or_else`](https://doc.rust-lang.org/nightly/std/result/enum.Result.html#method.or_else)

Do we also want to have a code example for this function somewhere? (I'm not sure what the rest of this crate does).

Reply:
Updated the docstring to include those things and an example.